cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -839,130 +846,114 @@ func buildScaleUpStatusNodeGroup(isScaleUpInProgress bool, scaleUpSafety NodeGro
 		condition.Status = api.ClusterAutoscalerUnhealthy
 	} else if !scaleUpSafety.SafeToScale {
 		condition.Status = api.ClusterAutoscalerBackoff
+		condition.BackoffInfo = api.BackoffInfo{
+			ErrorCode:    scaleUpSafety.BackoffStatus.ErrorInfo.ErrorCode,
+			ErrorMessage: scaleUpSafety.BackoffStatus.ErrorInfo.ErrorMessage,
```

Comment:
I wonder if we should maybe trim this to some maximum length? Max size of configmap is 1MB and that's not much if you have a few hundred nodegroups. 
A few hundred characters should likely be a reasonable limit for an error message?

Reply:
I actually don't print all the error messages. Only the first three error messages in each node group are printed.
They are also the same error messages used in events.

I added truncation logic for extra safety, PTAL.